webview/platforms/android/__init__.py

```diff
@@ -1,150 +1,57 @@
-        request = DownloadManagerRequest(uri)
-        request.setNotificationVisibility(visibility)
-        request.setDestinationInExternalFilesDir(context, dir_type, filepath)
-        dm = cast(DownloadManager, activity.getSystemService(Context.DOWNLOAD_SERVICE))
-        dm.enqueue(request)
-
-
-class KeyListener(PythonJavaClass):
-    __javacontext__ = 'app'
-    __javainterfaces__ = ['android/view/View$OnKeyListener']
-
-    def __init__(self, listener):
-        super().__init__()
-        self.listener = listener
-
-    @java_method('(Landroid/view/View;ILandroid/view/KeyEvent;)Z')
-    def onKey(self, v, key_code, event):
-        if key_code == KeyEvent.KEYCODE_BACK:
-            return self.listener()
-
-
-class EventCallbackWrapper(PythonJavaClass):
-    __javacontext__ = 'app'
-    __javainterfaces__ = ['com/pywebview/EventCallbackWrapper']
-
-    def __init__(self, callback):
-        super().__init__()
-        self.callback = callback
-
-    @java_method('(Ljava/lang/String;Ljava/lang/String;)V')
-    def callback(self, event, data):
-        if self.callback:
-            self.callback(event, data)
-
-
-class JsApiCallbackWrapper(PythonJavaClass):
-    __javacontext__ = 'app'
-    __javainterfaces__ = ['com/pywebview/JsApiCallbackWrapper']
-
-    def __init__(self, callback):
-        super().__init__()
-        self.callback = callback
-
-    @java_method('(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V')
-    def callback(self, func, params, id):
-        if self.callback:
-            Runnable(self.callback)(func, params, id)
-
-
-class RequestInterceptor(PythonJavaClass):
-    __javacontext__ = 'app'
-    __javainterfaces__ = ['com/pywebview/WebViewRequestInterceptor']
-
-    def __init__(self, webview_instance):
-        super().__init__()
-        self.webview = webview_instance
-
-    @java_method('(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;')
-    def onRequest(self, url: str, method: str, headers_json: str):
-        headers = json.loads(headers_json) if headers_json else {}
-        original_headers = headers.copy()
-
-        request = Request(url, method, headers)
-        self.webview.pywebview_window.events.request_sent.set(request)
-
-        if request.headers != original_headers:
-            logger.debug('Request headers mutated. Original: %s, Mutated: %s', original_headers, request.headers)
-
-            return json.dumps(request.headers)
-
-        return None
-
-    @java_method('(Ljava/lang/String;ILjava/lang/String;)V')
-    def onResponse(self, url: str, status_code: int, headers_json: str):
-        headers = json.loads(headers_json) if headers_json else {}
-
-        response = Response(url, status_code, headers)
-        self.webview.pywebview_window.events.response_received.set(response)
-
-
-def run_ui_thread(f, *args, **kwargs):
-    Runnable(f)(args, kwargs)
-
-
-class BrowserView(Widget):
-    def __init__(self, window, **kwargs):
+class BrowserView:
```

Comment:
As BrowserView no longer inherits from Widget, `get_size` method fails

Reply:
It is called from a `Window` object

Reply:
Fixed in the recent commit